session_0/word_processing.rb

```diff
@@ -2,13 +2,21 @@
 # any non-lower case alphabet (A..Z) to corresponding lower case
 # alphabet
 def lower_case(words)
-  raise NotImplementedError # TODO
+  array = Array.new
+  words.each do |word|
+    array.append(word.downcase)
+  end
+  array
 end
 
 # Similar to `lower_case`, this function modifies the array in-place
 # and does not return any value.
 def lower_case!(words)
-  raise NotImplementedError # TODO
+  i=0
+  while i< words.length() do
```

Comment:
Interesting that you have used `each` to iterate over arrays in `lower_case` but not here - why?